actionpack/lib/action_dispatch/middleware/stack.rb

```diff
@@ -34,7 +34,28 @@ def inspect
       end
 
       def build(app)
-        klass.new(app, *args, &block)
+        InstrumentationProxy.new(klass.new(app, *args, &block), inspect)
+      end
+    end
+
+    # This class is used to instrument the execution of a single middleware.
+    # It proxies the `call` method transparently and instruments the method
+    # call.
+    class InstrumentationProxy
+      EVENT_NAME = "process_middleware.action_dispatch"
```

Comment:
We can use the string literal directly; no need for a constant.

Reply:
It was done more for code clarity purposes...